client/skills.go

```diff
@@ -0,0 +1,146 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Skill represents a capacity offered by a snap.
+type Skill struct {
+	Name  string                 `json:"name"`
+	Snap  string                 `json:"snap"`
+	Type  string                 `json:"type,omitempty"`
+	Attrs map[string]interface{} `json:"attrs,omitempty"`
+	Apps  []string               `json:"apps,omitempty"`
+	Label string                 `json:"label,omitempty"`
+}
+
+// Slot represents the potential of a given snap to use a skill.
+type Slot struct {
```

Comment:
Ah, no reason at all. I'll make it so. I assume I can also do that in the
skills package.

2016-02-01 13:51 GMT+01:00 John Lenton <notifications@github.com>:

> In client/skills.go
> <https://github.com/ubuntu-core/snappy/pull/386#discussion_r51414210>:
>
> > +	"bytes"
> > +	"encoding/json"
> > +)
> > +
> > +// Skill represents a capacity offered by a snap.
> > +type Skill struct {
> > +	Name  string                 `json:"name"`
> > +	Snap  string                 `json:"snap"`
> > +	Type  string                 `json:"type,omitempty"`
> > +	Attrs map[string]interface{} `json:"attrs,omitempty"`
> > +	Apps  []string               `json:"apps,omitempty"`
> > +	Label string                 `json:"label,omitempty"`
> > +}
> > +
> > +// Slot represents the potential of a given snap to use a skill.
> > +type Slot struct {
>
> why isn't Slot just an alias to Skill? (assuming you need it to be a type
> of its own to keep the api clear)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snappy/pull/386/files#r51414210>.
>